spec/ddtrace/contrib/rails/rack_spec.rb

```diff
@@ -215,6 +251,95 @@ def internal_server_error
+      expect(controller_span).to have_error
+      expect(controller_span).to have_error_type('ActionView::MissingTemplate')
+      expect(controller_span).to have_error_message(include('Missing template test/does_not_exist'))
+      expect(controller_span).to have_error_stack
+
+      expect(template_span.name).to eq('rails.render_template')
+      expect(template_span.resource).to eq('rails.render_template') # Fallback value due to missing template
+      expect(template_span.span_type).to eq('template')
+      expect(template_span.get_tag('rails.template_name')).to be_nil
+      expect(template_span.get_tag('rails.layout')).to be_nil
+      expect(template_span).to have_error
+      expect(template_span).to have_error_type('ActionView::MissingTemplate')
+      expect(template_span).to have_error_message(include('Missing template test/does_not_exist'))
+      expect(template_span).to have_error_stack
+    end
+  end
+
+  context 'trying to render a nonexistent partial template' do
+    subject(:response) { get '/nonexistent_partial' }
+
+    before do
+      skip 'Event-based instrumentation cannot suffer from this issue' if Rails.version >= '4.0.0'
+    end
+
+    it 'traces complete stack' do
+      is_expected.to be_server_error
+
+      expect(spans).to have(4).items
+
+      request_span, controller_span, partial_span, template_span = spans
+
+      expect(request_span.name).to eq('rack.request')
+      expect(request_span.resource).to eq('TestController#nonexistent_partial')
+      expect(request_span).to have_error
+      expect(request_span).to have_error_type('ActionView::Template::Error')
+      expect(request_span).to have_error_message(include('Missing partial test/no_partial_here'))
+      expect(request_span).to have_error_stack
+
+      expect(controller_span.name).to eq('rails.action_controller')
+      expect(controller_span.resource).to eq('TestController#nonexistent_partial')
+      expect(controller_span).to have_error
+      expect(controller_span).to have_error_type('ActionView::Template::Error')
+      expect(controller_span).to have_error_message(include('Missing partial test/no_partial_here'))
+      expect(controller_span).to have_error_stack
+
+      expect(template_span.name).to eq('rails.render_template')
+      expect(template_span.resource).to eq('partial_does_not_exist.html.erb') # Fallback value due to missing template
+      expect(template_span.span_type).to eq('template')
+      expect(template_span.get_tag('rails.template_name')).to eq('partial_does_not_exist.html.erb')
+      expect(template_span.get_tag('rails.layout')).to eq('layouts/application')
+      expect(template_span).to have_error
+      expect(template_span).to have_error_type('ActionView::Template::Error')
+      expect(template_span).to have_error_message(include('Missing partial test/no_partial_here'))
+      expect(template_span).to have_error_stack
+
+      expect(partial_span.name).to eq('rails.render_partial')
```

Comment:
Minor: It's somewhat confusing to follow the expects not being in the same order as we get from spans; e.g. I think it'd be easier to read if they go from request_span --> template_span or in the reverse direction, rather than jumping around.

Reply:
Matched order with `spans` response.

Reply:
PR is much larger now because of it, but it seems easier to read 😅